api/serializers/assets.py

```diff
@@ -130,6 +140,39 @@ def to_internal_value(self, data):
             internals["specs"] = specs
         return internals
 
+    # # allow updating of active_inactive and paid for specific asset_types
+    def update(self, instance, validated_data):
+
+        asset_type = instance.model_number.asset_make.asset_type.name
+        paid = validated_data.get('paid')
+        active_inactive = validated_data.get('active')
+        expiry_date = validated_data.get('expiry_date')
+        instance_type_and_associated_error = {
+            "simcard": {"paid": "Only sim cards can have this field updated"},
+            "mifi": {"active": "Only mifi cards can be activated or deactivated"},
+            "embursecard": {
+                "expiry_date": "Only emburse cards can have this field updated"
+            },
+        }
+
+        if paid or expiry_date or (active_inactive is False or active_inactive is True):
```

Comment:
`active_inactive is False or active_inactive is True` this will always return True as long as `active_inactive` is a boolean value.

There is no need for `active_inactive is False or active_inactive is True`

Just do this to catch when active_inactive is None
```
if paid or expiry_date or active_inactive
```

Reply:
@raheemazeezabiodun That is not part of this PR. I have a commit from develop that has mixed up with my commits...

Reply:
Alright

Although that part needs to be fixed, can you include it to your pr?

Reply:
@raheemazeezabiodun No